frame/staking/src/lib.rs

```diff
@@ -917,3 +917,19 @@ impl BenchmarkingConfig for TestBenchmarkingConfig {
 	type MaxValidators = frame_support::traits::ConstU32<100>;
 	type MaxNominators = frame_support::traits::ConstU32<100>;
 }
+
+/// Means for checking if there is any external restriction on bonding with a specific accounts
+///
+/// Allows for parts of the runtime that might implement other forms fund locking to prevent
+/// incompatible locking on accounts which could lead to unsafe state.
+pub trait BondingRestriction<AccountId> {
+	/// Determine if bonding is allowed with stash and controller combination
```

Comment:
We are of course mainly concerned with the stash account, I added controller to give ability to prevent bond in-case the controller had some other lock on it which would prevent user from unbonding or doing other actions with their controller account. Obviously I didn't see the `set_controller`. (And coming to think of it, even if we detected it earlier, nothing would prevent user from using their controller account to stake so it would not have really prevented that anyway) I will drop controller as suggested.